### PR TITLE
SW-5493 Added table density to TableWithSearchFilters component

### PR DESCRIPTION
**Pull request by tommylau523**

### CODE REVIEW

**Comment by tommylau523** [comment]
* **#2711** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2711?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tommylau523 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/components/TableWithSearchFilters/index.tsx [hunk 1]
```diff
@@ -126,6 +128,16 @@ const TableWithSearchFilters = (props: TableWithSearchFiltersProps) => {
     });
   }, [extraTableFilters]);
 
+  const [tableDensity, setTableDensity] = useState<TableDensityType>();
```
Comment (joshbuchea) [inline src/components/TableWithSearchFilters/index.tsx:131]: Is it ok that `tableDensity` is `undefined` by default? Do `TableDensitySettingsButton` and `OrderPreservedTable` both provide a default value when `density` is `undefined`?

**Review by joshbuchea — APPROVED** [review]
I wonder how many tables will require the local state management for table density. If there are very many, it may be worth it to create a `useTableDensity` hook 🤔

**Comment by tommylau523** [comment]
### Merge activity

* **Jun 7, 4:25 PM EDT**: @tommylau523 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2711).

File: src/components/TableWithSearchFilters/index.tsx (hunk 1 above)
Comment (tommylau523) [reply src/components/TableWithSearchFilters/index.tsx:131]: It is ok! The BaseTable will load tableDensity from userPreference by default. The state management is just to speed it up by like half a second, to update the density settings before the BE can confirm it basically.

Comment (tommylau523) [reply src/components/TableWithSearchFilters/index.tsx:131]: Ideally we move every table to use the TableWithSearchFilter component, and manage the state in one place only.

**Pull request merged**